### PR TITLE
reworking drilling plant following 0lafe ticket

### PR DESCRIPTION
**Pull request by POPlol333**
Everything asked except speed, I don't feel like touching it honestly, scared of breaking too much the balance or just doing garbage

On this mod work is done, still has to modify and add controller craft in nhcore and modify the infinite pump craft to take a rig IV instead of a III

close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/8911

### CODE REVIEW

File: src/main/java/gregtech/loaders/preload/GT_Loader_MetaTileEntities.java [hunk 1]
```diff
@@ -1259,6 +1259,7 @@ private static void run3() {
         ItemList.OilDrill1.set(new GT_MetaTileEntity_OilDrill1(1157, "multimachine.oildrill1", "Oil/Gas/Fluid Drilling Rig").getStackForm(1));
         ItemList.OilDrill2.set(new GT_MetaTileEntity_OilDrill2(141, "multimachine.oildrill2", "Oil/Gas/Fluid Drilling Rig II").getStackForm(1));
         ItemList.OilDrill3.set(new GT_MetaTileEntity_OilDrill3(142, "multimachine.oildrill3", "Oil/Gas/Fluid Drilling Rig III").getStackForm(1));
+        ItemList.OilDrill3.set(new GT_MetaTileEntity_OilDrill4(149, "multimachine.oildrill4", "Oil/Gas/Fluid Drilling Rig IV").getStackForm(1));
```
Comment (Glease) [inline src/main/java/gregtech/loaders/preload/GT_Loader_MetaTileEntities.java:1262]: This is not going to work. Add a OilDrill4 in ItemList.

**Review by Glease — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/loaders/preload/GT_Loader_MetaTileEntities.java (hunk 1 above)
Comment (POPlol333) [reply src/main/java/gregtech/loaders/preload/GT_Loader_MetaTileEntities.java:1262]: oh wait didn't notice right, but how does that worked on my client lmao
thanks  btw

**Comment by Dream-Master** [comment]
Guess a recipe for control block need to be added

**Comment by POPlol333** [comment]
All the controller recipe are in nhcore (except the infinite which is in tectec) so I'll add the craft there

**Comment by Dream-Master** [comment]
sure

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilDrill3.java [hunk 2]
```diff
@@ -27,22 +27,22 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
...
     @Override
     protected Materials getFrameMaterial() {
-        return Materials.TungstenSteel;
+        return Materials.Titanium;
     }
 
     @Override
     protected int getCasingTextureIndex() {
-        return 48;
+        return 50;
```
Comment (boubou19) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilDrill3.java:40]: Same here

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilDrill4.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    protected Materials getFrameMaterial() {
+        return Materials.TungstenSteel;
+    }
+
+    @Override
+    protected int getCasingTextureIndex() {
+        return 48;
```
Comment (boubou19) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilDrill4.java:41]: same here

**Review by boubou19 — CHANGES_REQUESTED** [review]

**Comment by POPlol333** [comment]
if anyone tell me what that is and how it work then sure?

**Review by Glease — APPROVED** [review]
Code looks ok. We now just need to settle on the recipes..

**Comment by Dream-Master** [comment]
Recipe comes in core mod like the others

**Pull request merged**